### PR TITLE
SPLASH-26: GraphQL client and mocks

### PR DESCRIPTION
**Pull request by Pepegaga**

### CODE REVIEW

File: src/app/home/screens/home-screen.component.tsx [hunk 1]
```diff
@@ -22,19 +25,25 @@ export const HomeScreenComponent: React.FC = () => {
...
-        );
+        dispatch(MUSIC_ACTIONS.PLAY.TRIGGER());
     }, [dispatch]);
 
+    const handleRegister = () => {
+        dispatch(REGISTER.TRIGGER({ username: 'vlad', login: 'sfvb', password: '234342' }));
+    };
+
+    const handleSearch = async () => {
+        const result = await dispatch(SEARCH('Sunflower'));
```
Comment (TheLonelyAstronaut) [inline src/app/home/screens/home-screen.component.tsx]: WTF is this?)

File: src/app/player/sagas.ts [hunk 2]
```diff
@@ -28,12 +31,15 @@ export function* seekTo(action: ReturnType<typeof MUSIC_ACTIONS.SEEK_TO_POSITION
     yield call(RNTrackPlayer.seekTo, action.payload.position);
 }
 
-export function* playSaga(action: ReturnType<typeof MUSIC_ACTIONS.PLAY.TRIGGER>): SagaIterator {
+export function* playSaga(): SagaIterator {
+    const track = client.getTrack();
```
Comment (TheLonelyAstronaut) [inline src/app/player/sagas.ts]: You're passing required data to action

File: src/app/search/actions.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import { createAction } from 'typesafe-redux-helpers';
+
+export const SEARCH = createAction('[Search]', (payload: string) => payload);
```
Comment (TheLonelyAstronaut) [inline src/app/search/actions.ts]: Rename, we also have search in Liked

File: src/graphql/api.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        } else {
+            users.push(payload);
+        }
+    };
+
+    getTrack = (): Track => {
+        return tracks[0];
+    };
+
+    getCurrentPlaylist = (): Track[] => {
```
Comment (TheLonelyAstronaut) [inline src/graphql/api.ts]: getPlaylistByID

File: src/graphql/api.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        const user = await users.filter((user) => user.username === payload.username);
+
+        if (user.length > 0) {
+            throw new Error('user already exists');
+        } else {
+            users.push(payload);
+        }
+    };
+
+    getTrack = (): Track => {
```
Comment (TheLonelyAstronaut) [inline src/graphql/api.ts]: Remove useless stuff

File: src/graphql/api.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        return tracks[0];
+    };
+
+    getCurrentPlaylist = (): Track[] => {
+        return tracks;
+    };
+
+    getPLaylists = (): Track[][] => {
```
Comment (TheLonelyAstronaut) [inline src/graphql/api.ts]: Rewrite using Playlist[]

File: src/mocks/playlists.ts [hunk 7]
```diff
@@ -0,0 +1,123 @@
+import { ArtworkType, Track } from '../types/music';
+
+export const playlist: Track[][] = [
```
Comment (TheLonelyAstronaut) [inline src/mocks/playlists.ts]: Playlist[]

File: src/mocks/users.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
+export interface userProps {
```
Comment (TheLonelyAstronaut) [inline src/mocks/users.ts]: UserProps

File: src/graphql/api.ts [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+                    authorization: token,
+                },
+            };
+        });
+
+        this.client.setLink(authLink.concat(httpLink));
+    };
+
+    login = async (payload: LoginPayload): Promise<AuthCompletedPayload> => {
+        const user = await users.filter((user) => user.username === payload.username);
```
Comment (TheLonelyAstronaut) [inline src/graphql/api.ts]: Check password?

File: src/App.tsx [hunk 10]
```diff
@@ -4,13 +4,16 @@ import FlashMessage from 'react-native-flash-message';
...
-import { RootRouterComponent } from './app/routing/root-router.component';
+import { RootRouterComponent } from './app/routing/components/root-router.component';
 import { useStore } from './app/store/use-store';
 import { SplashScreen } from './app/ui/splash-screen.component';
 import { ConnectedThemeProvider } from './app/ui/themes/components/connected-theme-provider.component';
+import { client } from './graphql/api';
 
 LogBox.ignoreAllLogs(true);
 
+console.log(client);
```
Comment (TheLonelyAstronaut) [inline src/App.tsx]: Remove this log

File: src/app/authentication/authentication.d.ts [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+export interface AuthCompletedPayload {
+    username: string;
+    token: string;
+}
+
+export interface RegisterCompletedPayload {
+    username: string;
+    login: string;
+    password: string;
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/authentication.d.ts]: >username
>>login
>>>password

x3 lul

File: src/app/search/search.state.ts [hunk 12]
```diff
@@ -0,0 +1,6 @@
+import { Track } from '../../types/music';
+
+export interface SearchState {
+    searchText: string;
+    results: Track[];
```
Comment (TheLonelyAstronaut) [inline src/app/search/search.state.ts]: results can be also artist or playlist, so create interface SearchResult, then results will be SearchResult[].

```
enum SearchResultType  {
    ARTIST,
    PLAYLIST,
    TRACK
}

type SearchResult = {
    data: Track | Artist | Playlist
    type: SearchResultType  
}
```

File: src/app/authentication/authentication.d.ts (hunk 11 above)
Comment (TheLonelyAstronaut) [reply src/app/authentication/authentication.d.ts]: BTW why we need this type?

File: src/graphql/api.ts [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+        } else {
+            console.log('registered');
+        }
+    };
+
+    // getPLaylistById = (id: number): Playlist[] => {
+    //     return playlist;
+    // };
+
+    search = async (name: string): Promise<Track[]> => {
```
Comment (TheLonelyAstronaut) [inline src/graphql/api.ts]: Change to SearchResult[]

File: src/app/authentication/authentication.d.ts (hunk 11 above)
Comment (Pepegaga) [reply src/app/authentication/authentication.d.ts]: because when we register we don't need token I guess

File: src/app/search/actions.ts [hunk 14]
```diff
@@ -0,0 +1,13 @@
+import { createAction } from 'typesafe-redux-helpers';
+
+import { Track } from '../../types/music';
+
+export interface SearchPayload {
+    text: string;
+    result: Track[];
```
Comment (TheLonelyAstronaut) [inline src/app/search/actions.ts]: SearchResult[]

**Pull request merged**